pkg/util/versions/versions.go

```diff
@@ -0,0 +1,43 @@
+package versions
+
+import (
+	"os"
+
+	"github.com/Masterminds/semver/v3"
+	"sigs.k8s.io/yaml"
+)
+
+const previewVersion = "preview"
+
+func Supported(path string) ([]*semver.Version, error) {
+	content, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	plainVersions := []struct {
+		Version string `json:"version"`
+	}{}
+	if err := yaml.Unmarshal(content, &plainVersions); err != nil {
+		return nil, err
+	}
+
+	var versions []*semver.Version
+	for _, v := range plainVersions {
+		if v.Version == previewVersion {
+			continue
+		}
+
+		ver, err := semver.NewVersion(v.Version)
+		if err != nil {
+			return nil, err
+		}
+		versions = append(versions, ver)
+	}
+
+	return versions, nil
+}
+
+func OldestUpgradableToLatest(version []*semver.Version) string {
+	return version[len(version)-1-2].String()
```

Comment:
This isn't quite right I don't think. Shouldn't it be something like `{current.major, current.minor-2, 0}`? This just returns the 3rd to last version in the list which at `2.6.3` could be `2.6.1`

Reply:
`versions.yaml` only has the latest for a minor version but https://kuma.io/docs/2.6.x/production/upgrades-tuning/upgrades/#upgrade-kuma explicitly says the patch version doesn't matter, so I suggest setting the patch of the returned version to `0`.

Reply:
@lobkovilya let's leave it like you have it and open a website PR to make the support policy explicit.